src/units/abilities.cpp

```diff
@@ -1264,14 +1265,49 @@ unit_ability_list attack_type::get_specials_and_abilities(const std::string& spe
 	unit_ability_list spe_list = get_specials(special);
 	if(!spe_list.empty()){
 		spe_list = overwrite_special_checking(special, spe_list, abil_list, "filter_self", true);
-		if(special == "plague" && !spe_list.empty()){
-			return spe_list;
-		}
 		abil_list.append(spe_list);
 	}
 	return abil_list;
 }
 
+unit_ability_list attack_type::get_specials_and_abilities_with_priority(const std::string& special) const
+{
+	//If `get_specials_and_abilities(special)` would return an empty list, returns `std::nullopt`.
+	//Otherwise, returns the highest priority item in the list.
+	//If two or more items are tied for highest priority, returns the one that was closer to the start of the list.
+	unit_ability_list abil_list = get_specials_and_abilities(special);
+	if(abil_list.empty()){
+		return abil_list;
+	}
+
+
+	std::vector<std::string> checking_string;
+	std::vector<config> lists = resources::gamedata->get_special_string_list();
```

Comment:
If there's more than one list for a given type, which list should win the next loop?

My worry here is, what if a multiplayer scenario and an era both provided such a list? What should be the rules for who provides the list?

Reply:
i will said the first to be loaded, but i understand and remove that.

Reply:
if i judge by 389:saved_game::expand_mp_events() in saved_game.cpp theorder should be: era, campaign and resource. scenario lodaded before all other(expand_scenario();).